wasmbinding/message_plugin.go

```diff
@@ -399,60 +394,52 @@ func (m *CustomMessenger) performSubmitAdminProposal(ctx sdk.Context, contractAd
 		}
 	}
 
-	if proposal.PinCodesProposal != nil {
-		p := proposal.PinCodesProposal
-		err := msg.SetContent(&wasmtypes.PinCodesProposal{
-			Title:       p.Title,
-			Description: p.Description,
-			CodeIDs:     p.CodeIDs,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to set content on PinCodesProposal")
-		}
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, errors.Wrap(err, "failed to validate incoming SubmitAdminProposal message")
 	}
 
-	if proposal.UnpinCodesProposal != nil {
-		p := proposal.UnpinCodesProposal
-		err := msg.SetContent(&wasmtypes.UnpinCodesProposal{
-			Title:       p.Title,
-			Description: p.Description,
-			CodeIDs:     p.CodeIDs,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to set content on UnpinCodesProposal")
-		}
+	response, err := m.Adminserver.SubmitProposalLegacy(sdk.WrapSDKContext(ctx), &msg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to submit proposal")
 	}
 
-	if proposal.UpdateAdminProposal != nil {
-		p := proposal.UpdateAdminProposal
-		err := msg.SetContent(&wasmtypes.UpdateAdminProposal{
-			Title:       p.Title,
-			Description: p.Description,
-			NewAdmin:    p.NewAdmin,
-			Contract:    p.Contract,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to set content on UpdateAdminProposal")
-		}
+	ctx.Logger().Debug("submit proposal legacy processed in msg server",
+		"from_address", contractAddr.String(),
+	)
+
+	return response, nil
+}
+
+func (m *CustomMessenger) performSubmitAdminProposal(ctx sdk.Context, contractAddr sdk.AccAddress, adminProposal *bindings.AdminProposal) (*admintypes.MsgSubmitProposalResponse, error) {
+	proposal := adminProposal
+	authority := authtypes.NewModuleAddress(admintypes.ModuleName)
+	var (
+		msg     *admintypes.MsgSubmitProposal
+		sdkMsgs []sdk.Msg
+		sdkMsg  sdk.Msg
+	)
+
+	cdc := m.AdminKeeper.Codec()
+	err := cdc.UnmarshalInterfaceJSON([]byte(proposal.ProposalExecuteMessage.Message), &sdkMsg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshall incoming sdk message")
 	}
 
-	if proposal.ClearAdminProposal != nil {
-		p := proposal.ClearAdminProposal
-		err := msg.SetContent(&wasmtypes.ClearAdminProposal{
-			Title:       p.Title,
-			Description: p.Description,
-			Contract:    p.Contract,
-		})
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to set content on ClearAdminProposal")
-		}
+	signers := sdkMsg.GetSigners()
+	if len(signers) != 1 {
+		return nil, errors.Wrap(sdkerrors.ErrorInvalidSigner, "should be 1 signer")
+	}
+	if !signers[0].Equals(authority) {
+		return nil, errors.Wrap(sdkerrors.ErrUnauthorized, "authority in incoming msg is not equal to admin module")
 	}
+	sdkMsgs = append(sdkMsgs, sdkMsg)
+	msg, err = admintypes.NewMsgSubmitProposal(sdkMsgs, contractAddr)
```

Comment:
you can simplify the func by removing the sdkMsgs variable and simply passing the sdkMsg in this method as
```golang
admintypes.NewMsgSubmitProposal([]sdk.Msg{sdkMsg}, contractAddr)
```